src/main/java/com/puppycrawl/tools/checkstyle/site/SiteUtil.java

```diff
@@ -269,4 +401,843 @@ public static String getParentModule(Class<?> moduleClass)
+                currentClass = currentClass.getSuperclass();
+            }
+        }
+
+        return result;
+    }
+
+    /**
+     * Constructs string with relative link to the provided document.
+     *
+     * @param moduleName the name of the module.
+     * @param document the path of the document.
+     * @return relative link to the document.
+     * @throws MacroExecutionException if link to the document cannot be constructed.
+     */
+    public static String getLinkToDocument(String moduleName, String document)
+            throws MacroExecutionException {
+        final Path templatePath = getTemplatePath(moduleName.replace("Check", ""));
+        if (templatePath == null) {
+            throw new MacroExecutionException(
+                    String.format(Locale.ROOT,
+                            "Could not find template for %s", moduleName));
+        }
+        final Path templatePathParent = templatePath.getParent();
+        if (templatePathParent == null) {
+            throw new MacroExecutionException("Failed to get parent path for " + templatePath);
+        }
+        return templatePathParent
+                .relativize(Paths.get(SRC, "xdocs", document))
+                .toString()
+                .replace(".xml", ".html")
+                .replace('\\', '/');
+    }
+
+    /** Utility class for extracting description from a method's Javadoc. */
+    private static final class DescriptionExtractor {
+
+        /** Variable that tells if we are in a CODE_LITERAL token. */
+        private static boolean isInCodeLiteral;
+        /** Variable that tells if we are in an HTML_ELEMENT token. */
+        private static boolean isInHtmlElement;
+        /** Variable that tells if we are in an HREF attribute. */
+        private static boolean isInHrefAttribute;
+
+        /**
+         * Extracts the description from the javadoc detail node. Performs a DFS traversal on the
+         * detail node and extracts the text nodes.
+         *
+         * @param javadoc the Javadoc to extract the description from.
+         * @param moduleName the name of the module.
+         * @return the description of the setter.
+         * @throws MacroExecutionException if the description could not be extracted.
+         * @noinspection TooBroadScope
+         * @noinspectionreason TooBroadScope - complex nature of method requires large scope
+         */
+        // -@cs[NPathComplexity] Splitting would not make the code more readable
+        // -@cs[CyclomaticComplexity] Splitting would not make the code more readable.
+        private static String getDescriptionFromJavadoc(DetailNode javadoc, String moduleName)
+                throws MacroExecutionException {
+            resetValues();
+            final StringBuilder description = new StringBuilder(128);
+            final Deque<DetailNode> queue = new ArrayDeque<>();
+            pushOntoQueue(queue, getDescriptionNodes(javadoc));
```

Comment:
No need to reinvent the wheel: ` getDescriptionNodes(j).forEach(queue::push);`

Reply:
need a reverse in here, too, missed this in example

Reply:
Done. No problem, I caught it.